### PR TITLE
agda.withPackages: use GHC with ieee754 as default

### PR DESCRIPTION
**Pull request by neosimsim**
###### Motivation for this change
As mentioned in the package description of ieee on Hackage,
ieee is deprecated in favor of ieee754.

###### Things done
Change default `ghcWithPackages` in `agda.withPackages`.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
    See "Hello World" in [nixos/tests/agda.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/agda.nix)
- [x] Ensured that relevant documentation is up to date

### CODE REVIEW

**Comment by neosimsim** [comment]
@turion told me that I should mention you, @alexarice.

**Comment by turion** [comment]
See http://hackage.haskell.org/package/ieee

**Review by turion — APPROVED** [review]

**Review by alexarice — APPROVED** [review]
Agda documentation seems to also recommend ieee754 now

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110501` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>agda</li>
</ul>
</details>

File: nixos/tests/agda.nix [hunk 1]
```diff
@@ -36,6 +36,8 @@ in
         "cp ${hello-world} HelloWorld.agda"
     )
     machine.succeed("agda -l standard-library -i . -c HelloWorld.agda")
+    # Check execution
+    machine.succeed('test "$(./HelloWorld)" = "Hello World!"')
```
Comment (neosimsim) [inline nixos/tests/agda.nix:40]: Maybe we should write the test like

```
assert "Hello World!" in machine.succeed(
  "./HelloWorld"
), "HelloWorld does not run properly"
```

which would be consistent with other test files.

Comment (alexarice) [reply nixos/tests/agda.nix:40]: Perhaps, I am not an expert on testing, but if it works and gives a nice error message then that is surely an improvement

Comment (neosimsim) [reply nixos/tests/agda.nix:40]: It doesn't make much of a difference. Using `assert` we also check that `./HelloWorld` returns 0. `test` only validates the output of `./HelloWorld`.

test:

```
machine: must succeed: test "$(./HelloWorld)" = "Hello World!"
machine: output:
error:
Traceback (most recent call last):
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 897, in run_tests
    exec(tests, globals())
  File "<string>", line 1, in <module>
  File "<string>", line 15, in <module>
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 422, in succeed
    raise Exception(
Exception: command `test "$(./HelloWorld)" = "Hello World!"` failed (exit code 1)
```

assert:

```
machine: must succeed: ./HelloWorld
(0.02 seconds)
error:
Traceback (most recent call last):
  File "/nix/store/0fhfz90b5vc4k4gsk0gd6rj88m7gllz5-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 897, in run_tests
    exec(tests, globals())
  File "<string>", line 1, in <module>
  File "<string>", line 15, in <module>
AssertionError: HelloWorld does not run properly
```

**Review by turion — APPROVED** [review]

**Review by alexarice — APPROVED** [review]

**Pull request merged**